src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.model.transaction;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+
+/**
+ * Represents a Transaction in the sales book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+
+public class Transaction {
+
+    private final String description;
+    private final BigDecimal value;
+    private final TxnStatus status;
+
+    /**
+     * Constuctor
+     * @param description description for txn
+     * @param value txn amount
+     */
+
+    public Transaction(String description, BigDecimal value) {
+        requireAllNonNull(description, value);
+        this.description = description;
+        this.value = value;
+        this.status = new TxnStatus();
+    }
+
+    /**
+     * Constructor with the additional specification of whether the transaction is closed.
+     */
+    private Transaction(String description, BigDecimal value, boolean isClosed) {
+        requireAllNonNull(description, value);
+        this.description = description;
+        this.value = value;
+        this.status = new TxnStatus(isClosed);
+    }
+
```

Comment:
To update in future iteration:
Allow for transaction to be closed, and have the TxnStatus be updated to reflect the new timestamp.